src/main/java/org/killbill/billing/plugin/notification/setup/EmailNotificationListener.java

```diff
@@ -221,10 +224,31 @@ private void sendEmailForPayment(final Account account, final ExtBusEvent killbi
         Preconditions.checkArgument(killbillEvent.getEventType() == ExtBusEventType.INVOICE_PAYMENT_FAILED || killbillEvent.getEventType() == ExtBusEventType.INVOICE_PAYMENT_SUCCESS, String.format("Unexpected event %s", killbillEvent.getEventType()));
 
         final Invoice invoice = osgiKillbillAPI.getInvoiceUserApi().getInvoice(invoiceId, context);
+
+        boolean oneTimePayment = false;
+        boolean recurringPayment = false;
+        for (InvoiceItem current : invoice.getInvoiceItems()) {
+            try {
+                if (current != null && current.getInvoiceItemType() != null) {
+                        if (current.getInvoiceItemType() == InvoiceItemType.EXTERNAL_CHARGE) {
+                            oneTimePayment = true;
+                        }
+                        if (current.getInvoiceItemType() == InvoiceItemType.RECURRING) {
+                            recurringPayment = true;
+                        }
+                }
+            } catch (Exception e) {
+                logService.log(LogService.LOG_WARNING, String.format("Failed to evaluate invoice items for: %s", killbillEvent.getAccountId()), e);
+            }
+        }
+
         if (invoice.getNumberOfPayments() == 0) {
-            // Aborted payment? Maybe no default payment method...
+            if (invoice.getBalance().equals(new BigDecimal(0)) && recurringPayment && !oneTimePayment) {
```

Comment:
use BigDecimal.compareTo() instead of equals(); equals() on BigDecimal includes scale as well as value, so 0 does not equal 0.0 does not equal 0.00 according to that method.